mlx/fast.cpp

```diff
@@ -94,11 +97,71 @@ array rms_norm(
   return fallback({x, weight})[0];
 }
 
+std::vector<array> RMSNorm::vjp(
+    const std::vector<array>& primals,
+    const std::vector<array>& cotangents,
+    const std::vector<int>& argnums,
+    const std::vector<array>& outputs) {
+  assert(primals.size() == 2);
+  assert(outputs.size() == 1);
+  assert(cotangents.size() == 1);
+
+  auto s = stream();
+  auto fallback = [eps = eps_, s](const std::vector<array>& inputs) {
```

Comment:
This is a genuine question because it's bending my brain a bit. But do you need to check the `argnums` in the fallback as well and only return the requested grads?

Reply:
Don't think so. The fallback implements the Primitive so it should return all of the outputs given the inputs.

Say for instance we want second order gradients. Then we 'll run the vjp of LayerNormVJP and get the gradients wrt all inputs (including the cotangent). Then we will select the one that was requested in argnums.